specreduce/wavelength_calibration.py

```diff
@@ -191,6 +191,10 @@ def model(self, new_model):
         self._clear_cache()
         self._model = new_model
 
+    @property
+    def fit_residuals(self):
+        return self._fit_resids
```

Comment:
should `_fit_resids` be initialized to `None` in the `__init__` to avoid an error when accessing this before calling `apply_to_spectrum`?  Should that case raise a useful error stating that `apply_to_spectrum` needs to be called first?

Reply:
My thought was that if it was not initialized, it WOULD raise an error if apply_to_spectrum wasn't run first.

I could do it this way - initialize it to None, have a check in the getter that if fit_residuals is None, raise a more informative error. Do you prefer that approach?

Reply:
Yes, but the error will seem cryptic to the user (just something like "WavelengthCalibration object has no attribute _fit_resids").  I think either what you suggest or the cached_property could both catch the error and suggest to the user what needs to be done from their end.

Reply:
i also added a test to make sure this error is raised when this happens

Reply:
would checking for `self._model is None` for the error get around the need to set `_fit_resids` to False?

Reply:
no, because both ._model and .model will always exist (default is a linear model), just not fit if .wcs isn't accessed.